### PR TITLE
software: faster alarms display

### PR DESCRIPTION
**Pull request by csommelet**

### CODE REVIEW

File: src/software/firmware/srcs/screen.cpp [hunk 1]
```diff
@@ -23,6 +23,12 @@
 
 // INITIALISATION =============================================================
 
+static const uint8_t MAX_ALARMS_DISPLAYED = 4;
+static const char* NO_ALARM_LINE = "PEAK  PLAT  PEEP    ";
+static const char* ALARM_LINE = "Alarm:              ";
+static const int ALARMS_CODE_POS = 6;
+static const int SCREEN_WIDTH = 20;
```
Comment (dsferruzza) [inline src/software/firmware/srcs/screen.cpp:30]: There is a `SCREEN_LINE_LENGTH` macro in `parameters.h`

File: src/software/firmware/srcs/screen.cpp [hunk 2]
```diff
@@ -91,18 +97,69 @@ void displaySettings(uint16_t peakPressureMax,
...
+    }
+
+    return hasChanged;
+}
+
 void displayAlarmInformation(uint8_t p_alarmCodes[], uint8_t p_nbTriggeredAlarms) {
-    screen.setCursor(0, 2);
-    screen.print("                    ");
-    screen.setCursor(0, 2);
+    if (!hasAlarmInformationChanged(p_alarmCodes, p_nbTriggeredAlarms)) {
```
Comment (csommelet) [inline src/software/firmware/srcs/screen.cpp:129]: There is a risk of data not being displayed as expected if the line is overwritten somewhere else in the code.

Comment (dsferruzza) [reply src/software/firmware/srcs/screen.cpp:129]: Indeed. Maybe put this sentence in a comment and let's accept the risks? 😛

Comment (csommelet) [reply src/software/firmware/srcs/screen.cpp:130]: done

**Pull request merged**